lib/signaling/v2/peerconnection.js

```diff
@@ -1499,7 +1499,9 @@ function shouldRecycleTransceiver(transceiver, pcv2) {
   return !transceiver.stopped && !pcv2._replaceTrackPromises.has(transceiver)
     && (transceiver.currentDirection === 'inactive'
       || transceiver.currentDirection === 'recvonly'
-      || transceiver.direction === 'recvonly');
+      || transceiver.direction === 'recvonly')
+    && transceiver.direction !== 'sendrecv'
+    && transceiver.direction !== 'sendonly';
```

Comment:
@makarandp0 Can't this be simplified to:
```suggestion
      || transceiver.direction === 'recvonly')
    && !transceiver.sender.track;
```
?

Reply:
@manjeshbhargav - I think your suggestion would work, but using direction/currentDirection makes the intent more explicit.

Reply:
also I am also not sure if `transceiver.sender.track`  would always be null. but if you think `transceiver.sender.track` is better indicator how about:

```
function shouldRecycleTransceiver(transceiver, pcv2) {
  return !transceiver.stopped && !pcv2._replaceTrackPromises.has(transceiver) && !transceiver.sender.track
}
```

Reply:
@makarandp0 I think the trouble here is that we are using `currentDirection`, which specifies the current transceiver direction. So, in this case, even though `direction` was set to `sendrecv`, it still got recycled because `currentDirection` is `recvonly`. So, I think we should just get rid of the dependence on `currentDirection`:
```js
return !transceiver.stopped
  && !pcv2._replaceTrackPromises.has(transceiver)
  && ['inactive', 'recvonly'].includes(transceiver.direction);
```